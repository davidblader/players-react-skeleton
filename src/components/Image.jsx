import React from 'react';
import PropTypes from 'prop-types';
// import images from '../img/*.png';

// const Image = ({ src, alt: '' }) =>
// (<img className="animal-crossing-img" src={images[src]} alt={alt}/>);

// parcel is not letting me import / bundle images :(
// use hacky css workaround :(
const Image = ({ src }) => (<div className={`animal-crossing-img ${src}`} />);

Image.propTypes = {
  src: PropTypes.string.isRequired,
};

export default Image;
