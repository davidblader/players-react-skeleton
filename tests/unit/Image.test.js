import React from 'react';
import { shallow } from 'enzyme';
import Image from '../../src/components/Image';

test('Image shallow renders', () => {
  const component = shallow(<Image />);
  expect(component).toMatchSnapshot();
});
