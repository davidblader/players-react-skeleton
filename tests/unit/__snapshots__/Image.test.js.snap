// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image shallow renders 1`] = `
<div
  className="animal-crossing-img gyroid"
/>
`;
